src/libraries/System.Text.Json/tests/Serialization/ContinuationTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class ContinuationTests
+    {
+        [Fact]
+        public static async Task ContinationAtNullShouldWork()
+        {
+            var stream = new MemoryStream();
+            {
+                var obj = new TestClass
+                {
+                    S = new string('x', 31),
+                    P = "",
+                    C = new()
+                    {
+                        S = "",
+                        P = null,
+                        C = null,
+                    }
+                };
+                await JsonSerializer.SerializeAsync(stream, obj);
+            }
+
+            stream.Position = 0;
+            {
+                // {"S":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","P":"","C":{"S":"","P":null,"C":null}}
+                //                                                                ^ continuation after 'u' (64 bytes)
+                var readOptions = new JsonSerializerOptions
+                {
+                    DefaultBufferSize = 64,
+                    IgnoreNullValues = true
+                };
+
+                var obj = await JsonSerializer.DeserializeAsync<TestClass>(stream, readOptions);
+
+                Assert.Equal(new string('x', 31), obj.S);
+                Assert.Equal("", obj.P);
+                Assert.NotNull(obj.C);
+                Assert.Equal("", obj.C.S);
+                Assert.Null(obj.C.P);
+                Assert.Null(obj.C.C);
+            }
+        }
+
+        private class TestClass
+        {
+            public string S { get; set; }
+            public string P { get; set; }
+            public TestClass C { get; set; }
```

Comment:
Can we add tests for when P & C are value types (one primitive, one complex/POCO), and also when they are nullable value types?

Also, it would be good to add tests for the above scenarios when `TestClass` is deserialized with a parameterized ctor. This should help inform whether any change is needed in [ReadJsonAsObject](https://github.com/dotnet/runtime/blob/7861772d39d134c4517dfa76ab92109b899de833/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfT.cs#L268).

```cs
private class TestClassWithParameterizedCtor
{
    public string S { get; }
    public string P { get; }
    public TestClassWithParameterizedCtor C { get; }

    public TestClassWithParameterizedCtor(string s, string p, TestClassWithParameterizedCtor c) =>
        (S, P, C) = (s, p, c);
}
```

Reply:
I added a lot more tests and hope they cover most of the relevant paths. Especially all tests are now testing continuations at every position of the payload and not only at the null token. `ReadJsonAsObject` seems to work as it is.